api/src/main/java/org/commonjava/indy/data/StoreDataManager.java

```diff
@@ -73,6 +73,13 @@
     ArtifactStore getArtifactStore( StoreKey key )
             throws IndyDataException;
 
+    /**
+     * Return the {@link ArtifactStore} instance corresponding to the given key, where key is a composite of {@link StoreType}
+     * (hosted, remote, or group) and name.
+     */
+    ArtifactStore getArtifactStore( StoreKey key, boolean forceQuery )
```

Comment:
I think we don't need to add this new method in interface level as it only serves for service data. Instead, just add it in ServiceStoreDataManager, and then call it with Class check and casting in code which needed. This will avoid to change other concrete classes to implement this.

Reply:
@ligangty Ok, amending in the new commit.

Reply:
@ligangty done